src/rabbit_types.erl

```diff
@@ -151,6 +151,8 @@
 
 -type(protocol_name() :: 'amqp0_8' | 'amqp0_9_1' | 'amqp1_0' | 'mqtt' | 'stomp' | any()).
 
+-type(node_type() :: 'disc' | 'disk' | 'ram').
```

Comment:
Why do we need to support two spellings of the word "disk"? I find it confusing, and it doesn't help pattern matching. If we want to support both where a user can type this, then the input should be converted to a single form immediately.